### PR TITLE
Grab.js weaker at a distance

### PR DESCRIPTION
**Pull request by PhilipRosedale**
When using mouse to grab, reduce energy of the grab as you get farther away and as objects get larger.  Will reduce the risk of accidentally knocking everything around at a distance.

### CODE REVIEW

**Comment by PhilipRosedale** [comment]
Test Plan:  

Use mouse to drag objects at distance around.  Notice that it should now get harder/slower as you are far away and as objects get really big.  Compare hand controller far grab motion to mouse motion - should be similar.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 664b54c004d4ca02d26346ba89573034c010e6bd available at: https://gist.github.com/f323c03a23f75c68562ceb3e9922a439<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-664b54c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-664b54c.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 664b54c004d4ca02d26346ba89573034c010e6bd available at: https://gist.github.com/f323c03a23f75c68562ceb3e9922a439<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-664b54c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-664b54c.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 10

No hand controller available so only tested with mouse.  Works as described and seems an improvement.  Hopefully someone with hand controller can test as well just to be sure.

File: scripts/system/controllers/grab.js [hunk 1]
```diff
@@ -48,6 +50,23 @@ var enabled = true;
...
     return "grab-" + MyAvatar.sessionUUID;
 }
+  
+var DISTANCE_HOLDING_ACTION_TIMEFRAME = 0.1; // how quickly objects move to their new position
+var DISTANCE_HOLDING_UNITY_MASS = 1200; //  The mass at which the distance holding action timeframe is unmodified
+var DISTANCE_HOLDING_UNITY_DISTANCE = 6; //  The distance at which the distance holding action timeframe is unmodified
+
+function distanceGrabTimescale(mass, distance) {
+    var timeScale = DISTANCE_HOLDING_ACTION_TIMEFRAME * mass /
+    DISTANCE_HOLDING_UNITY_MASS * distance /
```
Comment (sethalves) [inline scripts/system/controllers/grab.js]: the indentation here looks suspicous

**Comment by hifi-gustavo** [comment]
Build console output for commit: c60b0c7ef540d7b1320a38f3f7cd5fb65c5622b1 available at: https://gist.github.com/f323c03a23f75c68562ceb3e9922a439<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-c60b0c7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9108-c60b0c7.dmg

**Pull request merged**